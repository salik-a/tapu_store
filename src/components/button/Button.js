import React from "react";
import { ButtonContainer, ButtonText } from './ButtonStyled'

const Button = ({ onPress, primary, secondary, title, textColor, borderColor, disabled }) => {

    const color = disabled ? primary : secondary;
    return (
        <ButtonContainer testID="button-touchable" onPress={onPress} bgColor={color} borderColor={borderColor} disabled={disabled}>
            <ButtonText testID="button-title" textColor={textColor} >{title}</ButtonText>
        </ButtonContainer>
    );
};
export default Button;