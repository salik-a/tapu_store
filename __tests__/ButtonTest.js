import React from 'react'
import { fireEvent, render, waitFor } from '@testing-library/react-native'
import Button from "../src/components/button/Button"
test('should match with snapshot', () => {
    const comp = render(<Button primary='#BBC3CF' secondary='#E82223' textColor='white' borderColor='rgba(0, 0, 0, .0)' />);
    expect(comp).toMatchSnapshot();
});

test("should render title correctly", () => {
    const testTitle = "test";
    const comp = render(<Button title={testTitle} primary='#BBC3CF' secondary='#E82223' textColor='white' borderColor='rgba(0, 0, 0, .0)' />)

    const buttonText = comp.getByTestId("button-title").children[0]
    expect(buttonText).toBe(testTitle)


});

test("should trigger onPress", () => {
    const mockFunction = jest.fn()
    const comp = render(<Button onPress={mockFunction} primary='#BBC3CF' secondary='#E82223' textColor='white' borderColor='rgba(0, 0, 0, .0)' />)

    const buttonTouchable = comp.getByTestId("button-touchable")
    fireEvent(buttonTouchable, "press")

    expect(mockFunction).toBeCalled()
})

test("can't press button if disabled true", () => {
    const mockFunction = jest.fn()
    const comp = render(<Button onPress={mockFunction} primary='#BBC3CF' secondary='#E82223' textColor='white' borderColor='rgba(0, 0, 0, .0)' disabled={true} />)
    const buttonTouchable = comp.getByTestId("button-touchable")
    fireEvent(buttonTouchable, "press")
    expect(mockFunction).toBeCalledTimes(0)
})

