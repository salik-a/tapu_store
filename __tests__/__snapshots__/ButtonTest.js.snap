// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#E82223",
      "borderColor": "rgba(0, 0, 0, .0)",
      "borderRadius": 12,
      "borderWidth": 1,
      "height": 50,
      "justifyContent": "center",
      "marginVertical": 10,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "85%",
    }
  }
  testID="button-touchable"
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 14,
          "textAlign": "center",
        },
      ]
    }
    testID="button-title"
    textColor="white"
  />
</View>
`;
